griptape/drivers/sql/amazon_redshift_sql_driver.py

```diff
@@ -0,0 +1,106 @@
+import time
+from typing import Optional
+import boto3
+from griptape.drivers import BaseSqlDriver
+from attr import define, field
+
+
+@define
+class AmazonRedshiftSqlDriver(BaseSqlDriver):
+    database: str = field(kw_only=True)
+    cluster_identifier: str = field(default=None, kw_only=True)
+    workgroup_name: str = field(default=None, kw_only=True)
+    db_user: str = field(default=None, kw_only=True)
+    secret_arn: str = field(default=None, kw_only=True)
+    region_name: str = field(default="us-east-1", kw_only=True)
+    wait_sec: float = field(default=0.3, kw_only=True)
+    client = field(default=None, kw_only=True)
+
+    def __attrs_post_init__(self):
+        if self.client == None:
+            session = boto3.Session(region_name=self.region_name)
+            self.client = session.client("redshift-data")
+
+    @staticmethod
+    def _process_rows_from_records(records) -> list[list]:
+        rows = []
+        for r in records:
+            tmp = [c[list(c.keys())[0]] for c in r]
+            rows.append(tmp)
+        return rows
+
+    @staticmethod
+    def _process_cells_from_rows_and_columns(
+        columns: list, rows: list[list]
+    ) -> list[dict[str, any]]:
+        return [{column: r[idx] for idx, column in enumerate(columns)} for r in rows]
+
+    @staticmethod
+    def _process_columns_from_column_metadata(meta) -> list:
+        return [k["name"] for k in meta]
+
+    @staticmethod
+    def _post_process(meta, records) -> list[dict[str, any]]:
+        columns = AmazonRedshiftSqlDriver._process_columns_from_column_metadata(meta)
+        rows = AmazonRedshiftSqlDriver._process_rows_from_records(records)
+        return AmazonRedshiftSqlDriver._process_cells_from_rows_and_columns(
+            columns, rows
+        )
+
+    def execute_query(self, query: str) -> Optional[list[BaseSqlDriver.RowResult]]:
```

Comment:
The output here will always be a `list`, not an `Optional`.

Reply:
Good catch, I'll update this to return `None` if there are no rows, like the [SqlDriver](https://github.com/griptape-ai/griptape/blob/main/griptape/drivers/sql/sql_driver.py#L24)